x-pack/plugins/security_solution/server/endpoint/services/actions/create/index.ts

```diff
@@ -0,0 +1,289 @@
+      .getMessageSigningService()
+      .sign(fleetActionDoc);
+    const signedFleetActionDoc = {
+      ...fleetActionDoc,
+      signed: {
+        data: fleetActionDocSignature.data.toString('base64'),
+        signature: fleetActionDocSignature.signature,
+      },
+    };
+
+    // write actions to .fleet-actions index
+    try {
+      fleetActionIndexResult = await this.esClient.index<EndpointAction>(
+        {
+          index: AGENT_ACTIONS_INDEX,
+          body: signedFleetActionDoc,
+          refresh: 'wait_for',
+        },
+        {
+          meta: true,
+        }
+      );
+
+      if (fleetActionIndexResult.statusCode !== 201) {
+        throw new Error(fleetActionIndexResult.body.result);
+      }
+    } catch (e) {
+      // create entry in .logs-endpoint.action.responses-default data stream
+      // when writing to .fleet-actions fails
+      if (doesLogsEndpointActionsDsExist) {
+        await createFailedActionResponseEntry({
+          esClient: this.esClient,
+          doc: {
+            '@timestamp': moment().toISOString(),
+            agent: doc.agent,
+            EndpointActions: {
+              action_id: doc.EndpointActions.action_id,
+              completed_at: moment().toISOString(),
+              started_at: moment().toISOString(),
+              data: doc.EndpointActions.data,
+            },
+          },
+          logger,
+        });
+      }
+
+      throw e;
+    }
+
+    if (casesClient) {
+      // convert any alert IDs into cases
+      let caseIDs: string[] = payload.case_ids?.slice() || [];
+      if (payload.alert_ids && payload.alert_ids.length > 0) {
+        const newIDs: string[][] = await Promise.all(
+          payload.alert_ids.map(async (alertID: string) => {
+            const cases: CasesByAlertId = await casesClient.cases.getCasesByAlertID({
+              alertID,
+              options: { owner: APP_ID },
+            });
+            return cases.map((caseInfo): string => {
+              return caseInfo.id;
+            });
+          })
+        );
+        caseIDs = caseIDs.concat(...newIDs);
+      }
+      caseIDs = [...new Set(caseIDs)];
+
+      // Update all cases with a comment
+      if (caseIDs.length > 0) {
+        const targets = endpointData.map((endpt: HostMetadata) => ({
+          hostname: endpt.host.hostname,
+          endpointId: endpt.agent.id,
+        }));
+
+        await Promise.all(
+          caseIDs.map((caseId) =>
+            casesClient.attachments.add({
+              caseId,
+              comment: {
+                type: CommentType.actions,
+                comment: payload.comment || '',
+                actions: {
+                  targets,
+                  type: payload.command,
+                },
+                owner: APP_ID,
+              },
+            })
+          )
+        );
+      }
+    }
+
+    const body = returnActionIdCommands.includes(payload.command) ? { action: actionID } : {};
```

Comment:
renaming to `actionId` probably makes more sense here

Reply:
👍